handlers/authentication.go

```diff
@@ -924,15 +949,46 @@ func OpenIdCodeFlowHandler(w http.ResponseWriter, r *http.Request) bool {
 	}
 
 	cipherSessionData := aesGcm.Seal(aesGcmNonce, aesGcmNonce, sessionDataJson, nil)
-	authCookie := http.Cookie{
-		Name:     config.TokenCookieName + "-aes",
-		Value:    base64.StdEncoding.EncodeToString(cipherSessionData),
+	base64SessionData := base64.StdEncoding.EncodeToString(cipherSessionData)
+
+	// If resulting session data is large, it may not fit in one cookie. So, the resulting
+	// session data is broken in chunks and multiple cookies are used, as is needed.
+	sessionDataChunks := chunkString(base64SessionData, business.SessionCookieMaxSize)
+	for i, chunk := range sessionDataChunks {
+		var cookieName string
+		if i == 0 {
+			// Set a cookie with the regular cookie name with the first chunk of session data.
+			// This is for backwards compatibility
```

Comment:
It's also nice that in most use-cases, you still only ever have one cookie as opposed to also needing a `-chunks` cookie with a value of 1.

Reply:
or do you always still have a `-chunks` cookie? I think this implementation does. Oh, well, never mind :)

Reply:
It's always there!
But it would be a simple condition to add around these lines: https://github.com/kiali/kiali/pull/3898/files#diff-3dd5896ffb3bb3685544e6a40338180ed95a5864b929b908fd1a70ec7628b251R983-R991

I can put that condition if you prefer not having the `-chunks` with value "1" :-)

Reply:
You can leave it alone. "No biggie"